test/browser/fragments.test.js

```diff
@@ -1686,13 +1652,7 @@ describe('Fragment', () => {
 
 		expect(scratch.innerHTML).to.equal(htmlForTrue);
 		expectDomLogToBe(
-			[
-				'<div>34.insertBefore(#text, <div>3)',
-				'<div>4.remove()',
-				'<div>3.remove()',
-				'<div>.appendChild(#text)',
-				'<div>1.appendChild(<div>2)'
-			],
```

Comment:
Here, we are transition from a tree like:
```html
<div class="wrapper">
  <div>3</div>
  <div>4</div>
</div>
```
to
```html
<div>1</div>
<div>2</div>
```

Our diff matches the old `div.wrapper` with the new `<div>1` node. So it sees a div with multiple children transitioning to a div with a single text child. To handle this, it clears away the old children by setting the `textContent` of `div.wrapper` to the new single text child. That change is why you no longer see the calls to `.remove()`. Also since we set `textContent` to the new text child, you don't see the relevant `insertBefore` call of the text node. In one operation we unmount old children and set new children.

We do still call `unmount` on old children so any lifecycle methods are called, but no DOM is removed in those calls since setting `textContent` will take care of that.